File: Java/Node.java
/**
 * Class that represents information with a single element in a linked list.
 */
public class Node<E> {
    E item;
    Node<E> next;

    /**
     * Default constructor. Sets next pointer to null.
     * 
     * @param item data
     */
    public Node(E item) {
        this(item, null);
    }

    /**
     * Constructor that uses the specified {@code item} and {@code next} values.
     * 
     * @param item data
     * @param next pointer to next node
     */
    public Node(E item, Node<E> next) {
        this.item = item;
        this.next = next;
    }
}

File: src/main/javascript/linkedlist/Node.js
/**
 * Class that represents information with a single element in a singly linked list.
 */
export class Node {

    /**
     * Data for this node.
     */
    item;

    /**
     * Pointer to next node.
     */
    next;

    /**
     * Creates a Node with the specified item and next values. 
     * Next is null by default.
     * 
     * @param {*} item data
     * @param {*} [next] pointer to next node 
     */
    constructor(item, next = null) {
        this.item = item;
        this.next = next;
    }
}